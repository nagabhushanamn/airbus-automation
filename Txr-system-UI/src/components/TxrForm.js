
import React, { useState } from 'react';

import { connect } from 'react-redux';
import { txr } from '../actions/txr'

export const TxrForm = ({ onSubmit, txr, message }) => {
    const [amount, setAmount] = useState(100)
    const [fromAccNumber, setFromAccNumber] = useState('1')
    const [toAccNumber, setToAccNumber] = useState('2')
    const handleSubmit = e => {
        e.preventDefault();
        const formData = { amount, fromAccNumber, toAccNumber }
        if (onSubmit) {
            onSubmit({ value: formData })
        }
        txr(formData)
    };
    return (
        <div className="row">
            <div className="col-sm-8 col-md-8">
                <div className="card">
                    <div className="card-header">Txr Form</div>
                    <div className="card-body">
                        <form id="txr-form" onSubmit={e => handleSubmit(e)}>
                            <div className="form-group">
                                <label htmlFor="fromAccNumber">Amount</label>
                                <input
                                    id="amount"
                                    type="number"
                                    value={amount}
                                    className="form-control"
                                    onChange={e => setAmount(e.target.value)}
                                />
                            </div>
                            <div className="form-group">
                                <label htmlFor="fromAccNumber">From Account</label>
                                <input
                                    id="fromAccNumber"
                                    type="number"
                                    value={fromAccNumber}
                                    className="form-control"
                                    onChange={e => setFromAccNumber(e.target.value)}
                                />
                            </div>
                            <div className="form-group">
                                <label htmlFor="toAccNumber">To Account</label>
                                <input
                                    id="toAccNumber"
                                    type="number"
                                    value={toAccNumber}
                                    className="form-control"
                                    onChange={e => setToAccNumber(e.target.value)}
                                />
                            </div>
                            <button className="btn btn-dark">Txr</button>
                        </form>
                    </div>
                </div>
            </div>
        </div>

    );
};


const mapStateToProps = (state) => {
    return {
        message: state.txr.message || ''
    }
}
const mapDispatchToProps = { txr }

export default connect(mapStateToProps, mapDispatchToProps)(TxrForm);