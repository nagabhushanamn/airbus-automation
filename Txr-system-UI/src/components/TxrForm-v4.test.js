

import React from 'react'
import renderer from 'react-test-renderer';
import {TxrForm} from './TxrForm'

test('TxrForm snapshot', () => {
    const tree = renderer.create(<TxrForm />)
    expect(tree).toMatchSnapshot()
})
