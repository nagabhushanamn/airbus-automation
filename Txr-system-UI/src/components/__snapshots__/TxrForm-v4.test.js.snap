// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxrForm snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-5 col-md-5"
  >
    <hr />
    <hr />
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        Txr Form
      </div>
      <div
        className="card-body"
      >
        <form
          id="txr-form"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="fromAccNumber"
            >
              Amount
            </label>
            <input
              className="form-control"
              id="amount"
              onChange={[Function]}
              type="number"
              value={100}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="fromAccNumber"
            >
              From Account
            </label>
            <input
              className="form-control"
              id="fromAccNumber"
              onChange={[Function]}
              type="number"
              value="1"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="toAccNumber"
            >
              To Account
            </label>
            <input
              className="form-control"
              id="toAccNumber"
              onChange={[Function]}
              type="number"
              value="2"
            />
          </div>
          <button
            className="btn btn-dark"
          >
            Txr
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
